noise/conn.go

```diff
@@ -17,6 +17,8 @@ type Conn struct {
 	// handshake
 	config            *Config // configuration passed to constructor
 	hs                handshakeState
+	remotePub         [32]byte // remote static public key
```

Comment:
> Because the hs.rs is cleared when the handshake completes (c.hs.clear()) and the clearing clears up also the public key. I think it could be relaxed to only clear up the private key but better be safe than sorry.

So. Can we think of good reasons for `clear()` to clear the public keys? If not, I'd rather have us modify the behavior of `clear()` rather than adding new fields here :o

Reply:
The reasons to clear a key from memory is to avoid an attacker  being able to read it (either because he's on the same machine or by using heartbleed style attacks, etc).  Usually we clear a private key because that's what is most urgent to keep secret. 
In a far far fetched scenario, we might also want to clear up the static public key to be able to  deny later that we were not participating in the communication ? (if we load the key via Yubikey on a computer which got compromised just after we used it so it still has the memory kept ? ) 
Lol this seems way to far fetched from me. So if you also can't find a realistic scenario, then let me know and I'll modify `clear()` so it only clears the private key.

Reply:
yeah I'm not particularly interested in the deniability property which Noise doesn't provide in any case. I think a better idea would be to stop `clear()` from removing the public keys. If you can make this change I'll accept your PR! Thanks again for the initial PR and the changes :) you rock